video/out/wayland_common.c

```diff
@@ -1705,6 +1705,14 @@ static void feedback_presented(void *data, struct wp_presentation_feedback *fbac
     struct vo_wayland_feedback_pool *fback_pool = data;
     struct vo_wayland_state *wl = fback_pool->wl;
 
+    bool current_zero_copy = flags & WP_PRESENTATION_FEEDBACK_KIND_ZERO_COPY;
+    if (!wl->first_frame_shown || wl->last_zero_copy != current_zero_copy) {
+        MP_DBG(wl, "Presentation was done with %s.\n",
+                 current_zero_copy ? "direct scanout" : "a copy");
+        wl->last_zero_copy = current_zero_copy;
+        wl->first_frame_shown = true;
```

Comment:
I'm thinking that instead you can use `-1` as the "undefined" value for this state and get rid of `first_frame_shown`. You could also set it to `-1` if feedback is discarded for any reason.